lib/prop/options.rb

```diff
@@ -17,8 +17,27 @@ def self.build(options)
       raise RuntimeError.new("Invalid threshold setting") unless result[:threshold] > 0
       raise RuntimeError.new("Invalid interval setting")  unless result[:interval] > 0
 
+      if result.key?(:increment)
+        result[:increment] = result[:increment].to_i
+        raise RuntimeError.new("Invalid increment setting") unless result[:increment] > 0
+      end
+
+      if leaky_bucket.include?(result[:strategy])
+        result[:burst_rate] = result[:burst_rate].to_i
+        if result[:burst_rate] < result[:threshold]
+          raise RuntimeError.new("Invalid burst rate setting")
+        end
+
+        result[:strategy] = Prop::LeakyBucketStrategy
+      else
+        result[:strategy] = Prop::BaseStrategy
+      end
+
       result
     end
 
+    def self.leaky_bucket
```

Comment:
I was thinking of passing in / using a strategy instance that stores the options, but this can be an extra refactoring step ... let's not break everything :D